python/ingestion/census_to_bq.py

```diff
@@ -102,3 +105,43 @@ def write_population_by_race_to_bq(dataset, table_name, gcs_bucket, filename):
   except json.JSONDecodeError as err:
     msg = 'Unable to write to BigQuery due to improperly formatted data: {}'
     logging.error(msg.format(err))
+
+
+def write_household_income_to_bq(dataset, table_name, gcs_bucket, file_prefix):
+  """Fetches all SAIPE blobs from a GCS bucket and uploads to a single BQ table. Also does some preprocessing.
+
+     dataset: The BigQuery dataset to write to
+     table_name: The name of the BigQuery table to write to
+     gcs_bucket: The name of the GCS bucket to pull from
+     file_prefix: File name prefix used to identify which GCS blobs to fetch"""
+  client = storage.Client()
+  bucket = client.get_bucket(gcs_bucket)
+  saipe_blobs = client.list_blobs(gcs_bucket, prefix=file_prefix)
+
+  frames = [] 
+  for blob in saipe_blobs:
+    json_string = blob.download_as_string()
+    frame = pandas.read_json(json_string, orient='values')
+    frame.rename(columns=frame.iloc[0], inplace=True)
+    frame.drop([0], inplace=True)
+    frames.append(frame)
+
+  concat = pandas.concat(frames, ignore_index=True)
+  # The SAIPE API includes the query predicate columns, which are duplicates of their
+  # ALL_CAPS counterparts. Toss 'em.
+  concat.drop(columns=['state', 'county', 'time'], inplace=True)
+
+  # Add an upload timestamp. This assumes the Cloud Run service is running in us-central1.
```

Comment:
Can you reuse the append_dataframe_to_bq method from the bq utils? I see that you are having it infer the schema, which is what I am doing too. Maybe if the column_types argument is None in the util function, the schema can be inferred. 

I don't know what the col_modes argument is and its not documented. But it can probably also be None for inferred schemas.

Reply:
Yeah, I didn't reuse that function because I am inferring the schema as well. I also am using load_table_from_dataframe() directly instead of writing back and forth to json strings and didn't want to clobber any other pipelines. 

I can generalize it to accept autodetect schemas, do you mind if I just lift it from your draft? Otherwise I can leave as-is if you want to handle it in your PR.